### PR TITLE
BB-168: separate transition processor

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

**Review by KillianG — APPROVED** [review]
For what it's worth, LGTM :)

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "lifecycle_conductor": "node extensions/lifecycle/conductor/service.js",
     "lifecycle_bucket_processor": "node extensions/lifecycle/bucketProcessor/task.js",
     "lifecycle_object_processor": "node extensions/lifecycle/objectProcessor/task.js",
+    "lifecycle_transition_processor": "LIFECYCLE_OBJECT_PROCESSOR_TYPE=transition node extensions/lifecycle/objectProcessor/task.js",
```
Comment (naren-rajendran) [inline package.json]: rename it to `lifecycle_object_transition_processor` perhaps?

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "lifecycle_conductor": "node extensions/lifecycle/conductor/service.js",
     "lifecycle_bucket_processor": "node extensions/lifecycle/bucketProcessor/task.js",
     "lifecycle_object_processor": "node extensions/lifecycle/objectProcessor/task.js",
```
Comment (naren-rajendran) [inline package.json:13]: should we rename this to `lifecycle_object_expiration_processor` because the default is `expiration`?

Comment (rachedbenmustapha) [reply package.json:13]: let's not break backward compatibility.

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**